contrib/scripts/localnet.sh

```diff
@@ -139,15 +169,21 @@ else
 fi
 
 echo_info "Adding genesis accounts..."
-echo "$MNEMONIC" | $BINARY keys add validator --recover
-if $BINARY add-genesis-account $($BINARY keys show validator -a) $GENESIS_COINS; then
-  echo_success "Successfully added genesis accounts"
+
+val_key_name="validator"
+
+echo "$MNEMONIC" | $BINARY keys add $val_key_name --recover
+if $BINARY add-genesis-account $($BINARY keys show $val_key_name -a) $GENESIS_COINS; then
+  echo_success "Successfully added genesis account: $val_key_name"
 else
-  echo_error "Failed to add genesis accounts"
+  echo_error "Failed to add genesis account: $val_key_name"
 fi
 
+val_address=$($BINARY keys list | jq -r '.[] | select(.name == "validator") | .address')
+val_address=${val_address:-"nibi1zaavvzxez0elundtn32qnk9lkm8kmcsz44g7xl"}
```

Comment:
Sets the `val_address` programmatically using `nibid keys list` rather than defaulting to a hardcoded one. That way, if we change the `$MNEMONIC` global variable, the script will still work.